### PR TITLE
CDAP 19699 Feature/No Data Screen

### PR DESCRIPTION
**Pull request by adityadivami**
# TITLE
No Records Screen

## Description
This feature contains the "No Records Screen" when no data is available on the Grid Table

## PR Type
- [ ] Bug Fix
- [x] Feature
- [ ] Build Fix
- [ ] Testing
- [x] General Improvement
- [ ] Cherry Pick

## Links
Jira: [Jira issue #](fill-in.org)
https://cdap.atlassian.net/browse/CDAP-19699

## Test Plan
https://docs.google.com/spreadsheets/d/1UCc9Dv1Vi204_sYMGHOlqZ_DujiRYlm01c88CIKmkMk/edit#gid=717913110

## Screenshots
Screenshot 2022-10-18 at 9 05 51 AM

### CODE REVIEW

File: app/cdap/components/GridTable/index.tsx [hunk 1]
```diff
@@ -32,6 +32,8 @@ import { useStyles } from './styles';
 import { flatMap } from 'rxjs/operators';
 import { IExecuteAPIResponse, IRecords, IParams, IHeaderNamesList } from './types';
 import { IValues } from 'components/WrangleHome/Components/OngoingDataExploration/types';
+import NoRecordScreen from 'components/NoRecordScreen/index';
```
Comment (njbriggs) [inline app/cdap/components/GridTable/index.tsx]: You don't need to include `/index` when importing

File: app/cdap/components/NoRecordScreen/styles.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    marginTop: 20,
+  },
+  subHeaderMessage: {
+    fontSize: 16,
+    color: '#000000',
+  },
+  innerWrapper: {
+    textAlign: 'center',
+  },
+});
```
Comment (njbriggs) [inline app/cdap/components/NoRecordScreen/styles.ts]: Please add a newline at the end of the file

File: app/cdap/components/NoRecordScreen/types.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+export interface INoDataScreenProps {
+    title: React.ReactNode;
+    subtitle: React.ReactNode;
+  }
```
Comment (njbriggs) [inline app/cdap/components/NoRecordScreen/types.ts]: Here too

File: app/cdap/components/NoRecordScreen/styles.ts (hunk 2 above)
Comment (njbriggs) [reply app/cdap/components/NoRecordScreen/styles.ts]: Also, please use `styled-components` for all new styling

File: app/cdap/text/text-en.yaml [hunk 4]
```diff
@@ -2088,7 +2088,10 @@ features:
         distributed: Distributed
         localSandbox: Local Sandbox
     rulesmgmt: Rules
-
+  NoRecordScreen:
```
Comment (njbriggs) [inline app/cdap/text/text-en.yaml]: Please add strings under `DataPrep`: https://github.com/cdapio/cdap-ui/blob/develop/app/cdap/text/text-en.yaml#L463
or create a new top-level entry for the new Wrangler UI

Comment (njbriggs) [reply app/cdap/text/text-en.yaml]: Actually, let's create a new top-level header for the new UI

File: app/cdap/components/GridTable/index.tsx (hunk 1 above)
Comment (sivaRajanaDivami) [reply app/cdap/components/GridTable/index.tsx]: Done

File: app/cdap/components/NoRecordScreen/styles.ts (hunk 2 above)
Comment (sivaRajanaDivami) [reply app/cdap/components/NoRecordScreen/styles.ts]: 1. Please add a newline at the end of the file ---> DONE
2. Also, please use styled-components for all new styling ---> Sure, We will use styled-components for all new styling going forward

File: app/cdap/components/NoRecordScreen/types.ts (hunk 3 above)
Comment (sivaRajanaDivami) [reply app/cdap/components/NoRecordScreen/types.ts]: Done

File: app/cdap/text/text-en.yaml (hunk 4 above)
Comment (sivaRajanaDivami) [reply app/cdap/text/text-en.yaml]: Done. created top-level-header(WranglerNewUI)

**Review by seanfeld — APPROVED** [review]

**Review by njbriggs — APPROVED** [review]
Please squash to one commit before merging

**Review by de-lan — APPROVED** [review]

**Comment by seanfeld** [comment]
if you squash the commits we can merge this one

**Comment by seanfeld** [comment]
also please link the jira

**Pull request merged**